src/components/card-decks.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { Link } from 'gatsby';
+import DottedBox from '../components/icons/dotted-box';
+
+const FullCard = ({ card }) => (
+  <div className="card rounded shadow-sm p-2">
+    <Link to={card.fields.path}>
+      <DottedBox className="opacity-1 mt-3 ml-3" width="100" height="100" />
+    </Link>
+    <div className="card-body">
+      <h3 className="card-title balance-text">
+        <Link to={card.fields.path}>{card.frontmatter.title}</Link>
+      </h3>
+
+      <p className="card-text">{card.frontmatter.description}</p>
+
+      {card.children.map(child => (
+        <Link
+          key={child.fields.path}
+          to={child.fields.path}
+          className="btn btn-link btn-block text-left p-0"
+        >
+          {child.frontmatter.title}
+        </Link>
+      ))}
+    </div>
+  </div>
+);
+
+const SimpleCard = ({ card }) => (
+  <div className="card rounded shadow-sm p-2">
+    <div className="card-body">
+      <h3 className="card-title balance-text">
+        <Link className="stretched-link" to={card.fields.path}>
+          {card.frontmatter.title}
+        </Link>
+      </h3>
+
+      <p className="card-text">{card.frontmatter.description}</p>
+    </div>
+  </div>
+);
+
+const CardDecks = ({ cards, groupSize = 3, cardType = 'simple' }) => {
+  let groupedCards = [];
```

Comment:
I rewrote this algorithm a couple times, any ideas to make it better? I tried a strictly linear time version, but I felt it was a little harder to understand. Wondering if there is an Array prototype function that could help make this better.